packages/nextcloud/lib/src/webdav/client.dart

```diff
@@ -56,24 +54,22 @@ class WebDavClient {
     return response;
   }
 
-  String _constructPath([final String? path]) => [
-        rootClient.baseURL,
-        webdavBasePath,
-        if (path != null) ...[
-          path,
-        ],
-      ]
-          .map((part) {
-            while (part.startsWith('/')) {
-              part = part.substring(1);
-            }
-            while (part.endsWith('/')) {
-              part = part.substring(0, part.length - 1); // coverage:ignore-line
-            }
-            return part;
-          })
-          .where((final part) => part.isNotEmpty)
-          .join('/');
+  Uri _constructUri([final Uri? path]) {
+    assert(
+      path == null || path.path == '/' || !path.path.startsWith('/') || !path.path.endsWith('/'),
+      "The path should not start or end with a '/' unless indicating the root folder.",
+    );
+
+    final pathBuffer = StringBuffer(webdavBasePath);
+    if (path != null && path.path != '/') {
+      pathBuffer
+        ..write('/')
+        ..write(path.path);
+    }
+
+    final builder = UriBuilder.fromUri(rootClient.baseURL)..path = pathBuffer.toString();
```

Comment:
Just setting to path will break instances hosted at a subfolder. The baseURL might already contain a path.